### DISCUSSION TITLE
Please help me...new to GitHUb

### DISCUSSION THREAD

**Discussion by MMagicoder** [community/community#54448] [Other Feature Feedback, Questions, & Ideas] [answered]
### Select Topic Area

Question

### Body

Hello all!

I am new to GitHub. (I joined a few days ago.) I have managed to learn how to read the theme documentation as well as the general static website creation documentation from GitHUb and create a blog. However, I always run into an issue: either the blog content (articles) don't show up, or the theme doesn't show up but the articles do. I keep seeing something about "Ruby Gems", but I don't have or want that on my computer. I simply want to follow the directions, put the correct html and md codes and files into my repository and publish my blog.  Is there no theme that allows that? Does everything require Ruby or something else to be downloaded?

Thanks!

**Comment by hrishikesh-kocharekar-dev** [comment]
GitHub Pages uses Jekyll, a static site generator that requires Ruby and some additional Ruby Gems to be installed on your computer. However, you don't necessarily need to install Ruby and the Gems locally to use GitHub Pages.

You can choose to use a Jekyll theme that's already set up to work with GitHub Pages without needing to install anything. There are many Jekyll themes available on GitHub that you can use to quickly create a blog or website.

To use a Jekyll theme on GitHub Pages without installing anything, you can follow these steps:

Fork the repository of the Jekyll theme you want to use. You can find many Jekyll themes by searching for "jekyll-theme" on GitHub.

Rename the forked repository to `<username>.github.io` where `<username>` is your GitHub username.

Customize the `_config.yml` file to personalize the theme to your needs.

Add your content as Markdown files in the `_posts` directory.

Push the changes to your forked repository.

Your blog should now be live at `https://<username>.github.io`.

If you want to further customize your Jekyll theme, you can use GitHub's built-in online code editor to make changes to your files and preview them in your browser.

I hope this helps you get started with using Jekyll themes on GitHub Pages!

**Reply by MMagicoder** [reply]
Hey! I did all of that and everything works except my posts don't show up.
Would you mind looking at my code and see what's wrong? I think I've followed all the instructions.
https://github.com/MMagicoder/mmagicoder.github.io/

**Reply by MMagicoder** [reply]
Thanks so much! This helped me a lot!

**Comment by airtower-luna** [comment]
Your Jekyll build is failing, see the error message in the last build: https://github.com/MMagicoder/mmagicoder.github.io/actions/runs/4888130486/jobs/8725506149#step:4:20

Looking at the file mentioned there, the YAML front matter looks like this:

```yaml
---
layout: post
title: "How and Why Organizations should Create Email Accounts for Volunteers"
date: YYYY-MM-DD
---
```

Indeed "YYYY-MM-DD" is not a valid date (it's a date *format*). You need to list the correct date for your post there (e.g. 2023-05-05).

**Reply by MMagicoder** [reply]
Yesterday, it was the 4th, so I used that date and it never showed up. The same is happening today now that I am using the 2023-05-05 format.
1.) Maybe I should find the original post format in the sample and go with that?
2.) I deleted the folder with the samples, but I don't think I need to make a doc folders and put the posts in there, do I?
3.) Lastly, I can't get rid of the "quickstart" link in the navigation of my website no matter how many html files I delete it from. What's the deal with that?

**Reply by MMagicoder** [reply]
Is there anything wrong with the "posts", "default" or "archive" html files? They all lead to each other, but maybe they shouldn't?

**Reply by airtower-luna** [reply, marked as answer]
> The same is happening today now that I am using the 2023-05-05 format.

Where did you try that? I see a recent commit that *renamed the file*, but you need to edit the date information *within the file*, exactly [here](https://github.com/MMagicoder/mmagicoder.github.io/blob/fc2cc92b0528df631823d50e30d42c75e5d8b510/_posts/2023-05-05-volunteer-email.md?plain=1#L4). That's what the error message I linked is about.

**Reply by MMagicoder** [reply]
OMG thanks! This worked!

**Comment by MMagicoder** [comment]
Oh, I think I misunderstood.
I thought the date automatically updated itself as long as it was in the title.
Let's cross our fingers and hope this works.
Thanks, by the way, to all of you, for all of your help!

**Comment by MMagicoder** [comment]
Okay everybody. I'm back again. So, I tried to change my config file so that I can get little pictures on top of my posts, but nothing happens. My commit stays stuck. I tried to change the post.html layout from "archive" to single.
Is that smart?

**Reply by airtower-luna** [reply]
I recommend you make a separate thread for that, with a bit more detail (and link to the commit) and a title that mentions Jekyll. That way you have a much better chance of someone who can help seeing your question.

**Reply by MMagicoder** [reply]
Thanks!
